tests/unit_tests/utils/test_general.py

```diff
@@ -912,3 +912,24 @@ def test_is_valid_boolean_type():
     assert not gen._is_valid_boolean_type(np.int32, None)
     assert not gen._is_valid_boolean_type(np.float32, None)
     assert not gen._is_valid_boolean_type(np.str_, None)
+
+
+def test_is_utf8_file(tmp_test_directory):
```

Comment:
Note that it might be worth merging this test with the `test_read_file_encoded_in_utf_or_latin` test cause you can use the same files created on the fly to test both.

Reply:
Very good point. I've removed the separated test_is_utf8 function and merged the tests.